Add Get/Update methods for Properties API

abhide opened the pull request.

Commit:
Update sling version to v1.3.0

Why is new version required?
Pick up fixes to specify custom ResponseDecoder

Commit:
Add Get/Update methods for Properties API

michaelw commented on clients/splunk/properties.go:
```diff
@@ -0,0 +1,73 @@
+
+	body := strings.NewReader(fmt.Sprintf("value=%s", value))
+	resp, err := p.client.New().Post(
+			fmt.Sprintf("properties/%s/%s/%s", file, stanza, key)).Body(body).ResponseDecoder(stringResponseDecoder{}).Receive(nil, apiError)
```
do the interpolated values need some kind of escaping?

abhide replied:
Fixed

michaelw commented on clients/splunk/properties.go:
```diff
@@ -0,0 +1,73 @@
+	apiError := &APIError{}
+	output := &Entry{}
+	resp, err := p.client.New().Get(
+		fmt.Sprintf("properties/%s/%s/%s", file, stanza, key)).ResponseDecoder(stringResponseDecoder{}).Receive(output, apiError)
```
ditto, see above

abhide replied:
Fixed

michaelw commented on clients/splunk/properties.go:
```diff
@@ -0,0 +1,73 @@
+	output := &Entry{}
+	resp, err := p.client.New().Get(
+		fmt.Sprintf("properties/%s/%s/%s", file, stanza, key)).ResponseDecoder(stringResponseDecoder{}).Receive(output, apiError)
+	if err != nil || !apiError.Empty() {
```
hmm, maybe we should make it such that we can check for exactly one condition.

abhide replied:
Will address these in a cleanup/refactor separate PR.

Commit:
Use go modules

Commit:
Ignore error from go clean -testcache

CI would fail with

make test TESTREPORT=test-results/go/results.xml
sed -e 's;@@GOBIN@@;/go/bin;g' < vault.hcl.in > vault.hcl
mkdir -p test-results/go/
go clean -testcache
go clean -testcache: open /tmp/go/cache/testexpire.txt: no such file or directory
make: *** [Makefile:29: test] Error 1

Commit:
Don't checkout repo under GOPATH

michaelw commented on Makefile:
```diff
@@ -31,18 +26,12 @@ dev: build
 	@test -n "$$SPLUNK_ADDR" || { echo 'warning: SPLUNK_ADDR not set, creating new Splunk instances.  This will be slow.'; }
 	mkdir -p $(dir $(TESTREPORT))
-	gotestsum --junitfile $(TESTREPORT) -- -cover -v ./...
+	go clean -testcache || true
```
```env GOCACHE=off gotestsum...```

abhide replied:
```$ make test
mkdir -p ./
env GOCACHE=off gotestsum --junitfile test-results.xml --format standard-verbose -- -cover -v ./...
build cache is disabled by GOCACHE=off, but required as of Go 1.12

=== Errors
build cache is disabled by GOCACHE=off, but required as of Go 1.12

DONE 0 tests, 1 error in 0.015s
make: *** [test] Error 1```
